timemachine/ff/handlers/nonbonded.py

```diff
@@ -608,14 +611,29 @@ def __init__(self, patterns, params, protein_ff_name, water_ff_name, topology):
         self.signs = np.array(signs)
 
     def parameterize(self, params):
+        # If there aren't any matched parameters (i.e. it's all water),
+        # then there is nothing to do
+        if len(self.param_idxs) == 0:
+            return self.initial_charges
+
         bond_deltas = params[self.param_idxs] * self.signs
         final_charges = apply_bond_charge_corrections(
             self.initial_charges, self.bond_idxs, bond_deltas, runtime_validate=False
         )
-
         return final_charges
 
 
+class EnvironmentBCCPartialHandler(SerializableMixIn):
```

Comment:
Updated

```This class is used to represent the environment BCC terms
that modify the charges used for the environment-ligand
interaction potential.

The current implementation skips water molecules, so only the
protein charges are modified.

This class is a serializable version of `EnvironmentBCCHandler`,
so it can be stored in the forcefield python file in the normal way.
```